### PR TITLE
Initial commit for NiFi Docs

### PR DESCRIPTION
**Pull request by aheyne**
This has been built and tested.

Signed-off-by: Austin Heyne <aheyne@ccri.com>

### CODE REVIEW

File: docs/user/nifi.rst [hunk 1]
```diff
@@ -0,0 +1,179 @@
+GeoMesa NiFi Plugins
+====================
+
+NiFi allows for managing large batches and streams of files and data.
+GeoMesa-NiFi allows you to ingest data into GeoMesa strait from NiFi by
```
Comment (mdzimmerman) [inline docs/user/nifi.rst]: strait -> straight

File: docs/user/nifi.rst [hunk 2]
```diff
@@ -0,0 +1,179 @@
...
+    $ mvn clean install
+
+
+To install the GoeMesa processors you will need to copy the geomesa-nifi-nar file from
+``geomesa-nifi/geomesa-nifi-nar/target/geomesa-nifi-nar-$VERSION.nar`` into the ``lib/`` directory of your NiFi installation.
+
+Install the SFTs and Converters
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+The GeoMesa processors need access to the SFTs and Converters in order to ingest data. We can provide these to the
```
Comment (mdzimmerman) [inline docs/user/nifi.rst]: [picky style] I don't think we need to capitalize "Converters"

Comment (mdzimmerman) [inline docs/user/nifi.rst]: Also, can we put a reference to the converters chapter, which talks more about the format of the reference.conf file?

File: docs/user/nifi.rst [hunk 3]
```diff
@@ -0,0 +1,179 @@
...
+processors by placing the SFTs and Converters in a file named ``reference.conf`` and then putting that file on the
+classpath. This can be achieved by wrapping this file in a jar and placing it in the ``lib/`` directory of the NiFi
```
Comment (mdzimmerman) [inline docs/user/nifi.rst]: [picky style] jar -> JAR

File: docs/user/nifi.rst [hunk 4]
```diff
@@ -0,0 +1,179 @@
...
+* PutGeoMesa       - Ingest data into GeoMesa Accumulo with a GeoMesa converter or from geoavro
+* PutGeoMesaKafka  - Ingest data into GeoMesa Kafka with a GeoMesa converter or from geoavro
+* PutGeoTools      - Ingest data into an arbitrary GeoTools Datastore based on parameters using a GeoMesa converter or avro
+* ConvertToGeoAvro - Use a GeoMesa converter to create geoavro
+
+PutGeoMesa
+^^^^^^^^^^
+
+The ``PutGeoMesa`` processor is used for ingesting data into an Accumulo backed GeoMesa datastore. To use this processor
+first add it to the workspace and open the properties tab of it's configuration. Descriptions of the properties are
```
Comment (mdzimmerman) [inline docs/user/nifi.rst]: it's -> its

File: docs/user/nifi.rst [hunk 5]
```diff
@@ -0,0 +1,179 @@
+GeoMesa NiFi Plugins
+====================
+
+NiFi allows for managing large batches and streams of files and data.
```
Comment (mdzimmerman) [inline docs/user/nifi.rst]: How about "NiFi manages large batches and streams of files and data." ?

File: docs/user/nifi.rst [hunk 6]
```diff
@@ -0,0 +1,179 @@
...
+useMock                       Use a mock instance of Accumulo
+GeoMesa Configuration Service Configuration service to use. More about this feature below.
+============================= ==========================================================================================
+
+GeoMesa Configuration Service
+"""""""""""""""""""""""""""""
+
+The ``PutGeoMesa`` plugin supports `NiFi Controller Services <http://docs.geoserver.org/stable/en/user/tutorials/cql/cql_tutorial.html>`__
+to manage common configurations. This allows the user to specify a single location to store the Accumulo connection parameters.
+This allows you to add new PutGeoMesa processors without having to enter duplicate datea.
```
Comment (mdzimmerman) [inline docs/user/nifi.rst]: datea -> data

File: docs/user/nifi.rst [hunk 7]
```diff
@@ -0,0 +1,179 @@
...
+
+To use this feature, after configuring the service, select the appropriate Geomesa Config Controller Service from the drop down
+of the ``GeoMesa Configuration Service`` property. When a controller service is selected the ``zookeepers``, ``instanceId``,
+``user``, ``password`` and ``tableName`` parameters are not required or used.
+
+PutGeoMesaKafka
+^^^^^^^^^^^^^^^
+
+The ``PutGeoMesaKafka`` processor is used for ingesting data into a Kafka backed GeoMesa datastore. To use this processor
+first add it to the workspace and open the properties tab of it's configuration. Descriptions of the properties are
```
Comment (mdzimmerman) [inline docs/user/nifi.rst]: it's -> its

File: docs/user/nifi.rst [hunk 8]
```diff
@@ -0,0 +1,179 @@
...
+cleanUpCache                  Run a thread to clean up the live feature cache if set to true. False by default. Use
+                              'cleanUpCachePeriod' to configure the length of time between cache cleanups. Every second
+                              by default.
+============================= ==========================================================================================
+
+PutGeoTools
+^^^^^^^^^^^
+
+The ``PutGeoTools`` processor is used for ingesting data into a GeoTools compatible datastore. To use this processor
+first add it to the workspace and open the properties tab of it's configuration. Descriptions of the properties are
```
Comment (mdzimmerman) [inline docs/user/nifi.rst]: it's -> its

File: docs/user/nifi.rst [hunk 9]
```diff
@@ -0,0 +1,179 @@
...
+============================= ==========================================================================================
+
+This processor also accepts dynamic parameters that my be needed for the specific datastore that you're trying to access.
+
+ConvertToGeoAvro
+^^^^^^^^^^^^^^^^
+
+The ``ConvertToGeoAvro`` processor leverages GeoMesa's internal converter framework to convert features into Avro and pass them
+along as a flow to be used by other processors in NiFi. To use this processor first add it to the workspace and open
+the properties tab of it's configuration. Descriptions of the properties are given below:
```
Comment (mdzimmerman) [inline docs/user/nifi.rst]: it's -> its

File: docs/user/nifi.rst [hunk 10]
```diff
@@ -0,0 +1,179 @@
...
+
+The ``PutGeoMesa`` processor is used for ingesting data into an Accumulo backed GeoMesa datastore. To use this processor
+first add it to the workspace and open the properties tab of it's configuration. Descriptions of the properties are
+given below:
+
+============================= ==========================================================================================
+Property                      Description
+============================= ==========================================================================================
+Mode                          Converter or Avro file ingest mode switch.
+SftName                       Name of the SFT on the classpath to use. This property override SftSpec.
```
Comment (mdzimmerman) [inline docs/user/nifi.rst]: override -> overrides   (this appears in the three other tables below as well)

File: docs/user/nifi.rst [hunk 11]
```diff
@@ -0,0 +1,179 @@
...
+FeatureNameOverride           Override the feature name on ingest.
+SftSpec                       SFT specification String. Overwritten by SftName if SftName is valid.
+ConverterSpec                 Converter specification string. Overwritten by ConverterName if ConverterName is valid.
+instanceId                    Accumulo instance ID
+zookeepers                    Comma separated list of zookeeper IPs or hostnames
+user                          Accumulo username with create-table and write permissions
+password                      Accumulo password for given username
+visibilities                  Accumulo scan visibilities
+tableName                     Name of the table to write to. If using namespaces but sure to include that in the name.
+writeTreads                   Number of threads to use when writing data to GeoMesa, has a linear effect on CPU and
```
Comment (mdzimmerman) [inline docs/user/nifi.rst]: is "writeTreads" right?

File: docs/user/nifi.rst [hunk 12]
```diff
@@ -0,0 +1,179 @@
...
+ConverterName                 Name of converter on the classpath to use. This property overrides ConverterSpec.
+FeatureNameOverride           Override the feature name on ingest.
+SftSpec                       SFT specification String. Overwritten by SftName if SftName is valid.
+ConverterSpec                 Converter specification string. Overwritten by ConverterName if ConverterName is valid.
+instanceId                    Accumulo instance ID
+zookeepers                    Comma separated list of zookeeper IPs or hostnames
+user                          Accumulo username with create-table and write permissions
+password                      Accumulo password for given username
+visibilities                  Accumulo scan visibilities
+tableName                     Name of the table to write to. If using namespaces but sure to include that in the name.
```
Comment (mdzimmerman) [inline docs/user/nifi.rst]: but sure -> be sure

File: docs/user/nifi.rst [hunk 13]
```diff
@@ -0,0 +1,179 @@
...
+Mode                          Converter or Avro file ingest mode switch.
+SftName                       Name of the SFT on the classpath to use. This property override SftSpec.
+ConverterName                 Name of converter on the classpath to use. This property overrides ConverterSpec.
+FeatureNameOverride           Override the feature name on ingest.
+SftSpec                       SFT specification String. Overwritten by SftName if SftName is valid.
+ConverterSpec                 Converter specification string. Overwritten by ConverterName if ConverterName is valid.
+DataStoreName                 Name of the datastore to ingest data into.
+============================= ==========================================================================================
+
+This processor also accepts dynamic parameters that my be needed for the specific datastore that you're trying to access.
```
Comment (mdzimmerman) [inline docs/user/nifi.rst]: my -> may

File: docs/user/nifi.rst (hunk 5 above)
Comment (aheyne) [inline docs/user/nifi.rst]: fixed

File: docs/user/nifi.rst (hunk 1 above)
Comment (aheyne) [inline docs/user/nifi.rst]: fixed

File: docs/user/nifi.rst (hunk 3 above)
Comment (aheyne) [inline docs/user/nifi.rst]: fixed

File: docs/user/nifi.rst (hunk 4 above)
Comment (aheyne) [inline docs/user/nifi.rst]: fixed

File: docs/user/nifi.rst (hunk 10 above)
Comment (aheyne) [inline docs/user/nifi.rst]: fixed

File: docs/user/nifi.rst (hunk 12 above)
Comment (aheyne) [inline docs/user/nifi.rst]: fixed

File: docs/user/nifi.rst (hunk 11 above)
Comment (aheyne) [inline docs/user/nifi.rst]: fixed

File: docs/user/nifi.rst (hunk 6 above)
Comment (aheyne) [inline docs/user/nifi.rst]: fixed

File: docs/user/nifi.rst (hunk 7 above)
Comment (aheyne) [inline docs/user/nifi.rst]: fixed

File: docs/user/nifi.rst (hunk 13 above)
Comment (aheyne) [inline docs/user/nifi.rst]: fixed

File: docs/user/nifi.rst (hunk 9 above)
Comment (aheyne) [inline docs/user/nifi.rst]: I swear I know the difference... fixed.

File: docs/user/nifi.rst (hunk 8 above)
Comment (aheyne) [inline docs/user/nifi.rst]: fixed

**Comment by aheyne** [comment]
Moved docs to geomesa-nifi and added import script

**Comment by jahhulbert-ccri** [comment]
@aheyne @mdzimmerman  looks good to me. I assume you tested it out? I'll read over the nifi docs PR which I assume i'll find in geomesa-nifi :+1:

**Pull request merged**